src/utils/dates.js

```diff
@@ -49,61 +46,68 @@ export const handleNextPurchaseDate = (
 };
 
 function getDaysBetweenDates(earlierDate, laterDate) {
+	/**
+	 * Calculate the number of days between two dates.
+	 * @param {Date} earlierDate The starting date.
+	 * @param {Date} laterDate The ending date.
+	 * @returns {number} The number of days between the two dates.
+	 */
 	return Math.floor(
 		(laterDate.getTime() - earlierDate.getTime()) / ONE_DAY_IN_MILLISECONDS,
 	);
 }
 
 function calculatePurchaseIntervals(
+	/**
+	 * Calculate the purchase intervals between current, next, and last purchase dates.
+	 * @param {Date} currentDate The current date.
+	 * @param {Date} dateNextPurchased The previously estimated next purchase date.
+	 * @param {Date|null} dateLastPurchased The date the item was last purchased (can be null).
+	 * @returns {Object} An object containing the last estimated interval and days since last purchase.
+	 */
 	currentDate,
 	dateCreated,
 	dateNextPurchased,
 	dateLastPurchased,
 ) {
-	const lastPurchaseDate = dateLastPurchased?.toDate(); // valid to be null
-	console.log(
-		`Previously purchased on: ${lastPurchaseDate === undefined ? undefined : lastPurchaseDate.toLocaleString()}`,
-	);
-	console.log(
-		`Previous next purchase date: ${dateNextPurchased.toDate().toLocaleString()}`,
-	);
+	const lastPurchaseDate = dateLastPurchased?.toDate();
+
+	const lastEstimatedIntervalStartDate =
+		lastPurchaseDate ?? dateCreated.toDate();
 
-	// valid to be undefined
 	const lastEstimatedInterval = getDaysBetweenDates(
-		lastPurchaseDate === undefined ? dateCreated.toDate() : lastPurchaseDate,
+		lastEstimatedIntervalStartDate,
 		dateNextPurchased.toDate(),
 	);
 
-	console.log(`Last estimated interval: ${lastEstimatedInterval}`);
-
-	// has to be a number
 	const daysSinceLastPurchase =
 		lastPurchaseDate === undefined
 			? 0
 			: getDaysBetweenDates(lastPurchaseDate, currentDate);
 
-	console.log(`Days since last purchase: ${daysSinceLastPurchase}`);
-
 	return { lastEstimatedInterval, daysSinceLastPurchase };
 }
 
 function getNextPurchaseEstimate(purchaseIntervals, totalPurchases) {
-	console.log(`Total purchases: ${totalPurchases}`);
+	/**
+	 * Calculate the next purchase estimate based on purchase intervals and total purchases.
+	 * @param {Object} purchaseIntervals The intervals between the previous and current purchases.
+	 * @param {number} purchaseIntervals.lastEstimatedInterval The previously estimated number of days between purchases.
+	 * @param {number} purchaseIntervals.daysSinceLastPurchase The number of days since the last purchase.
+	 * @param {number} totalPurchases The total number of purchases made.
+	 * @returns {Date} The estimated next purchase date.
+	 * @throws {Error} If an error occurs during the next purchase estimation process.
+	 */
```

Comment:
same here, move comment above function declaration

Reply:
great catch! not sure how they got moved inside the bodies of functions, I wonder if that could have been prettier